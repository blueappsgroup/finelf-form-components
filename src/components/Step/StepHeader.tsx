import styled from 'styled-components'
import { device } from '../../consts/sizes'

type StepHeaderProps = {
  activeStep?: boolean
  theme: {
    stepHeaderActiveColor: string
    stepHeaderInActiveColor: string
    stepHeaderActiveBorderColor: string
    stepHeaderInActiveBorderColor: string
  }
}

const StepHeader = styled.div<StepHeaderProps>`
  display: flex;
  justify-content: center;
  flex-grow: 1;
  font-size: 16px;
  letter-spacing: 0px;
  font-weight: ${(props: StepHeaderProps): number =>
    props.activeStep ? 800 : 400};
  color: ${(props: StepHeaderProps): string =>
    props.activeStep
      ? props.theme.stepHeaderActiveColor
      : props.theme.stepHeaderInActiveColor};
  border-bottom: ${(props: StepHeaderProps): string =>
    props.activeStep
      ? `4px solid ${props.theme.stepHeaderActiveBorderColor}`
      : `2px solid ${props.theme.stepHeaderInActiveBorderColor}`};
  margin: 0px 0px 30px;
  padding-bottom: 10px;

  &:first-of-type {
    margin-right: 15px;
  }

  &:last-child {
    margin-left: 15px;
  }

  @media ${device.mobileL} {
    &:first-of-type {
      margin-right: 15px;
    }

    &:last-child {
      margin-left: 15px;
    }
  }
`

export default StepHeader
