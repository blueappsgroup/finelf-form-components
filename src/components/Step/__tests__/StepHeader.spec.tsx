import * as React from 'react'
import { render } from '@testing-library/react'
import StepHeader from '../StepHeader'

describe('<StepHeader />', () => {
  it('matches snapshot', () => {
    const documentBody = render(<StepHeader />)
    const { baseElement } = documentBody
    expect(baseElement.querySelector('div')).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })
})
