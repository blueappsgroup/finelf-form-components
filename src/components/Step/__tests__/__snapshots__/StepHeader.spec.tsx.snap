// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepHeader /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bdnylx kUhWLa"
    />
  </div>
</body>
`;
